packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -449,17 +449,37 @@ extern dec useSystemTextJsonConverter(target: Model, scope?: valueof string);
  *   blobName: string;
  * }
  *
- * @@clientInitialization(MyService, MyServiceClientOptions)
+ * @@clientInitialization(MyService, {parameters: MyServiceClientOptions})
  * // The generated client will have `blobName` on it. We will also
  * // elevate the existing `blobName` parameter to the client level.
  * ```
  */
 extern dec clientInitialization(
   target: Namespace | Interface,
-  options: Model,
+  options: ClientInitializationOptions,
   scope?: valueof string
 );
 
+/**
+ * Client initialization options.
+ */
+model ClientInitializationOptions {
+  /**
+   * Redefine the client initialization parameters.
+   */
+  parameters?: Model;
+
+  /**
+   * Determines the accessibility of the client initialization method. `Access.public` means the client could be initialized separately.
+   */
+  access?: EnumMember;
+
+  /**
+   * Determines the accessibility of the client accessor method. `Access.public` means client could be got from parent client.
+   */
+  accessorAccess?: EnumMember;
```

Comment:
i don't think this name is very clear, but I don't have a great suggestion. Accessible from parent? I'm just not sure that a spec author will have an understanding of "accessor" like us tcgc authors do

Reply:
will change to use `initailizedBy`